global-script.lua

```diff
@@ -363,9 +377,18 @@ function onLoad(saved_data)
                     o.interactable = false -- sets boards to uninteractable after reload
                 end
             end
+        else
+            enableUI()
+            if BnCAdded then SetupChecker.UI.hide("bnc") end
```

Comment:
these expansion checks shouldn't be needed with latest changes anymore

Reply:
Which changes?  Without this tiny bugfix, you could add the expansions and then undo and the setup UI would show the buttons to add them again.  Adding them a second time would cause it to bug out and then think the cards were still being added for eternity without ever finishing.  This is the one way the setup UI would actually break (instead of simply losing state) across an undo/reload.

Reply:
https://github.com/spirit-island/spirit-island-mod/commit/c07d1fb313ee30ac0751ef1663e90f5b2b00364e#diff-43229af7b500c76924eddd5e569e67402f210fff5c0601f11190ddd8b9b591cf

Reply:
I'll have to look into this more.  It still looked like I was able to reproduce the issue at the time, but maybe it is actually fixed now.

Reply:
Still occurs in beta.  Repro:

1. Load initial save
2. Add B&C and Jagged Earth
3. Wait 12-15 seconds for the Undo timer and Undo *or* Save/Load
4. Note the B&C and Jagged Earth buttons are back
5. Click them again and receive an error.

Incidentally, under my fork I'm seeing the buttons briefly go invisible before re-appearing and I'm not entirely certain why.  I'll check against clean beta and see if I can reproduce the above -- but my code only hides, not shows, so something weird is definitely going on there.

Reply:
Yup, still occurs in clean beta.

Reply:
well the reason why they would reappear is because of how dropdowns work. There is no clean way of updating things currently so instead you have to update the entire setup UI and change some text. this leads to some race conditions. regardless though, this change has nothing to do with the actual PR